<?xml version="1.0" ?>
<robot name="bogson" xmlns:xacro="http://www.ros.org/wiki/xacro">
  
  <!-- BGN - Include -->
  <xacro:include filename="$(find lab2)/bogson_template.urdf.xacro" />
  <xacro:include filename="$(find lab2)/parameters.urdf.xacro" />
  <!-- END - Include -->

  <!-- BGN - Robot description -->
  <xacro:m_link name="base"
    originxyz="0 0 0"
    dimensions="5 5 1">
  </xacro:m_link>

  <xacro:m_link name="link1"
    originxyz="0 0 ${a1/2}"
    dimensions="1 1 ${a1}">
  </xacro:m_link>

  <xacro:joint name="base_to_link1"
    type="revolute"
    axis="0 0 1"
    originrpy="0 0 0"
    originxyz="0 0 1"
    parent="base" 
    child="link1">
  </xacro:joint>

  <xacro:m_link name="link2"
    originxyz="0 0 ${a2/2}"
    dimensions="1 1 ${a2}">
  </xacro:m_link>

  <xacro:joint name="link1_to_link2" 
    type="revolute"
    axis="0 0 1"
    originrpy="0 0 ${1.57075}"
    originxyz="0 0 ${a1}"
    parent="link1"
    child="link2">
  </xacro:joint>

  <xacro:m_link name="link3"
    originxyz="0 0 ${a3/2}"  
    dimensions="1 1 ${a3}">
  </xacro:m_link>

  <xacro:joint name="link2_to_link3" 
    type="revolute"
    axis="0 0 1"
    originrpy="0 0 0"
    originxyz="0 0 ${a2}"
    parent="link2"
    child="link3">
  </xacro:joint>

  <xacro:m_link name="narzedzie"
    originxyz="0 0 0.5"
    dimensions = "1 1 1">
  </xacro:m_link>

  <xacro:joint name="link3_to_narzedzie" 
    type="revolute"
    axis="0 0 1"
    originrpy="0 0 0"
    originxyz="0 0 ${a3}"
    parent="link3"
    child="narzedzie">
  </xacro:joint>

  <!-- END - Robot description -->

</robot>
